keps/sig-storage/3751-volume-attributes-class/README.md

```diff
@@ -172,27 +172,48 @@ type PersistentVolumeClaimSpec struct {
   ...
 }
 
+// Add the PersistentVolumeClaimModifyVolume condition to existing PersistentVolumeClaimConditionType
+// Condition is used to document the last error controller sees
+const (
+  ...
+  // PersistentVolumeClaimVolumeAttriburesModifyError - a user trigger modify volume of pvc has error
+  PersistentVolumeClaimVolumeAttriburesModifyError PersistentVolumeClaimConditionType = "VolumeAttriburesModifyError"
+  ...
+)
+
+
 // PersistentVolumeClaimStatus represents the status of PV claim
 type PersistentVolumeClaimStatus struct {
   ...
-  // The VolumeAttributesClassName string cannot be empty
-  VolumeAttributesClassName string
-  ModifyVolumeStatus *PersistentVolumeClaimModifyVolumeStatus
+  // ModifyVolumeStatus represents the status object of ControllerModifyVolume operation
+  ModifyVolumeStatus ModifyVolumeStatus
   ...
 }
 
+// ModifyVolumeStatus represents the status object of ControllerModifyVolume operation
+type ModifyVolumeStatus struct {
+  // CurrentVolumeAttributesClassName is the current name of the VolumeAttributesClass the PVC is using
+  CurrentVolumeAttributesClassName string
+  // TargetVolumeAttributesClassName is the name of the VolumeAttributesClass the PVC currently being reconciled
+  TargetVolumeAttributesClassName string
+  // Status is the status of the ControllerModifyVolume operation
+  Status *PersistentVolumeClaimModifyVolumeStatus
```

Comment:
@sunnylovestiramisu @msau42

Why use a pointer type? Do we want to distinguish a `nil` value and a zero value? If so, why? If not, why not use a non-pointer type?

Reply:
History of this being a pointer: https://github.com/kubernetes/enhancements/pull/3780/files#r1230300561

Reply:
Hmm actually this should not be a pointer.